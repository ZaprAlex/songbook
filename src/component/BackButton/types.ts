export interface BackButtonProps {
    onClick?: () => void;
    dataTestId?: string;
    label?: string;
    adaptive?: boolean;
}
