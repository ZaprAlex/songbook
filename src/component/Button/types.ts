import React from 'react';

export type ButtonActionTypes = 'button' | 'submit' | 'reset';
export type ButtonTypes = 'contained' | 'outline' | 'text';
export type ButtonSizes = 'small' | 'middle' | 'large' | 'x-large';
export type ButtonTypeColor = 'green';
export type ButtonIconPosition = 'left' | 'right';

export interface IButton {
    id?: string;
    label?: string;
    color?: ButtonTypeColor;
    icon?: string | React.ReactNode;
    iconPosition?: ButtonIconPosition;
    htmlType?: ButtonActionTypes;
    type?: ButtonTypes;
    size?: ButtonSizes;
    adaptive?: boolean;
    skeuomorph?: boolean;
    justify?: boolean;
    className?: string;
    disabled?: boolean;
    loading?: boolean;
    onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    dataTestId?: string;
}
