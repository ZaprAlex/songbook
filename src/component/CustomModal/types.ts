export type CustomModalProps = Record<string, unknown>;
