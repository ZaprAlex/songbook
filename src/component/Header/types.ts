import { ReactNode } from 'react';

type Position = 'fixed' | 'sticky' | 'relative' | 'static';
type BackgroundColor = 'transparent' | 'white';

export interface HeaderProps {
    renderLeft?: () => ReactNode;
    renderCenter?: () => ReactNode;
    renderRight?: () => ReactNode;
    position?: Position;
    backgroundColor?: BackgroundColor;
    className?: string;
    htmlElement?: HTMLElement;
    onHelpButtonClick?: () => void;
    adaptive?: boolean;
}
