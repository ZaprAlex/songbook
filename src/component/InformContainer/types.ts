import { IButton } from '../Button';

export interface IInformContainer {
    icon?: string;
    header: string;
    headerClassName?: string;
    message?: string;
    messageClassName?: string;
    buttonLabel?: string;
    buttonCancelLabel?: string;
    contentClassName?: string;
    buttonClassName?: string;
    buttonCancelClassName?: string;
    onMount?: () => void;
    onClick?: () => void;
    onCancelClick?: () => void;
    buttonCancelProps?: IButton;
    buttonProps?: IButton;
    loading?: boolean;
}
