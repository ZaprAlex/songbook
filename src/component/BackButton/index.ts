export { default } from './BackButton';
export * from './types';
