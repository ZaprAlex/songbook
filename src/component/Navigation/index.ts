import NavigationProvider from './NavigationProvider';
import useAppNavigation, { AppNavigationCtx } from './NavigationContext';

export { NavigationProvider, AppNavigationCtx, useAppNavigation };
