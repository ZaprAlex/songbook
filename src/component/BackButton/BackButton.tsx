import React, { FC, useCallback } from 'react';

import { useAppNavigation } from '../Navigation';
import Button from '../Button';
import arrowLeft from '../../static/svgs/arrow-left-black.svg';

import { BackButtonProps } from './types';

const BACK_BUTTON_LABEL = 'Назад';

const BackButton: FC<BackButtonProps> = ({
    onClick,
    dataTestId = 'back-button',
    label = BACK_BUTTON_LABEL,
    adaptive,
}) => {
    const { goBack } = useAppNavigation();
    const backButtonClick = useCallback(() => {
        if (onClick) {
            onClick();
        } else {
            goBack();
        }
    }, [onClick]);

    return (
        <Button
            onClick={backButtonClick}
            label={label}
            adaptive={adaptive}
            type="outline"
            size="small"
            icon={arrowLeft}
            dataTestId={dataTestId}
        />
    );
};

export default BackButton;
