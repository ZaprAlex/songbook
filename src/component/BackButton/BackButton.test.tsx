import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';

import { AppNavigationCtx } from '../Navigation';

import BackButton from './BackButton';
import { BackButtonProps } from './types';

describe('BackButton ', () => {
    let onClickMock: () => void, navigationMock: any;

    beforeEach(() => {
        onClickMock = jest.fn();
        navigationMock = { goBack: onClickMock };
    });

    const component = (props?: BackButtonProps) =>
        mount(
            <AppNavigationCtx.Provider value={navigationMock}>
                <BackButton {...props} />
            </AppNavigationCtx.Provider>
        );

    it('Is func match snapshot', () => {
        const wrapper = component();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('onclick test w/o onClick prop', () => {
        const container = component();
        container.find('button').simulate('click');
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    it('onclick test with onClick prop', () => {
        const props = {
            onClick: onClickMock,
        };
        const container = component(props);
        container.find('button').simulate('click');
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});
