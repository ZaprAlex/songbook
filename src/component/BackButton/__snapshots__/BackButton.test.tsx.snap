// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton  Is func match snapshot 1`] = `
<BackButton>
  <Button
    dataTestId="back-button"
    icon="arrow-left-black.svg"
    label="Назад"
    onClick={[Function]}
    size="small"
    type="outline"
  >
    <button
      className="btn btn-icon-left btn-outline btn-small"
      data-test-id="back-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="btnIcon"
      >
        <WhiteSvgLogo
          url="arrow-left-black.svg"
        >
          <svg />
        </WhiteSvgLogo>
      </span>
      <span
        className="buttonLabel"
      >
        Назад
      </span>
    </button>
  </Button>
</BackButton>
`;
